### PR TITLE
Feat: Nu5 Tx (v5) updates

### PR DESCRIPTION
**Pull request by niklaslong**
See: https://zips.z.cash/protocol/protocol.pdf#networkupgrades

### CODE REVIEW

File: src/protocol/payload/tx.rs [hunk 1]
```diff
@@ -421,6 +424,236 @@ impl Codec for TxV4 {
...
+        self.actions_orchard.encode(buffer)?;
+
+        if !self.actions_orchard.is_empty() {
+            buffer.put_u8(self.flags_orchard.unwrap());
+            buffer.put_i64_le(self.value_balance_orchard.unwrap());
+            buffer.put_slice(&self.anchor_orchard.unwrap());
+
+            VarInt(self.proofs_orchard.as_ref().unwrap().len()).encode(buffer)?;
+            for proof in self.proofs_orchard.as_ref().unwrap() {
+                buffer.put_u8(*proof)
```
Comment (ljedrz) [inline src/protocol/payload/tx.rs:509]: `proofs_orchard` can probably be encoded "wholesale" here via `put_slice`

File: src/protocol/payload/tx.rs [hunk 2]
```diff
@@ -421,6 +424,236 @@ impl Codec for TxV4 {
...
+        let (
+            flags_orchard,
+            value_balance_orchard,
+            anchor_orchard,
+            proofs_orchard,
+            auth_sigs_orchard,
+            binding_sig_orchard,
+        ) = if !actions_orchard.is_empty() {
+            // Decode the orchard flags.
+            if bytes.remaining() < 1 {
```
Comment (ljedrz) [inline src/protocol/payload/tx.rs:585]:
```suggestion
            if bytes.remaining() == 0 {
```

File: src/protocol/payload/tx.rs [hunk 3]
```diff
@@ -785,4 +1125,37 @@ mod tests {
 
         assert_eq!(tx_v4, Tx::decode(&mut Cursor::new(&bytes)).unwrap());
     }
+
+    #[test]
+    #[ignore]
```
Comment (ljedrz) [inline src/protocol/payload/tx.rs:1130]: why do we ignore this? also, it would be best to test with a non-empty tx, preferably from some original test vector

**Review by ljedrz — APPROVED** [review]
LGTM, left a few comments.

File: src/protocol/payload/tx.rs (hunk 3 above)
Comment (niklaslong) [reply src/protocol/payload/tx.rs:1130]: We ignored all the unit tests so that we can run Ziggurat with just `cargo t`, though perhaps a feature flag would be more appropriate in the long run.

Comment (niklaslong) [reply src/protocol/payload/tx.rs:1128]: > preferably from some original test vector
Agreed, I'll follow up with this if I can find some.
